sklearn/metrics/scorer.py

```diff
@@ -32,10 +32,21 @@
 from ..base import is_regressor
 
 
+def _get_metric_name(scorer):
+    """Get a name for the predefined/custom metric function."""
+    if isinstance(scorer, _BaseScorer):
+        return scorer._name
+    elif getattr(scorer, '__name__', '') == '_passthrough_scorer':
+        return "default"
+    else:
+        return "custom"
+
+
 class _BaseScorer(six.with_metaclass(ABCMeta, object)):
-    def __init__(self, score_func, sign, kwargs):
+    def __init__(self, score_func, sign, name, kwargs):
```

Comment:
I hope you don't find this superfluous. I can't think of a more succinct way to get the scorer names without using `__name__` other than by explicitly specifying a `_name` attribute.